dv/cosim/spike_cosim.cc

```diff
@@ -440,6 +447,34 @@ void SpikeCosim::leave_nmi_mode() {
 #endif
 }
 
+void SpikeCosim::handle_cpuctrl_exception_entry() {
+  bool old_sync_exc_seen = change_cpuctrlsts_sync_exc_seen(true);
+
+  if (old_sync_exc_seen) {
+    set_cpuctrlsts_double_fault_seen();
+  }
+}
+
+bool SpikeCosim::change_cpuctrlsts_sync_exc_seen(bool flag) {
+  bool old_flag = false;
+  uint32_t cpuctrlsts = processor->get_csr(CSR_CPUCTRLSTS);
+
+  if (cpuctrlsts & 0x40) {
+    old_flag = true;
+  }
+
+  cpuctrlsts = (cpuctrlsts & 0x1bf) | (flag ? 0x40 : 0);
+  processor->put_csr(CSR_CPUCTRLSTS, cpuctrlsts);
+
+  return old_flag;
+}
+
+void SpikeCosim::set_cpuctrlsts_double_fault_seen() {
+  uint32_t cpuctrlsts = processor->get_csr(CSR_CPUCTRLSTS);
+  cpuctrlsts = (cpuctrlsts & 0x17f) | 0x80;
```

Comment:
Nit: put a comment here that double fault seen flag is bit 7.

Reply:
Also, is the mask can be `0x1ff`

Reply:
It can, but I prefer to leave it as 0x17f, makes it more explicit we're ignoring the value of double_fault_seen here